{
    "containers": {
        "cna": {
            "affected": [
                {
                    "product": "postgresql",
                    "vendor": "n/a",
                    "versions": [
                        {
                            "status": "affected",
                            "version": "postgresql 13.3, postgresql 12.7, postgresql 11.12"
                        }
                    ]
                }
            ],
            "descriptions": [
                {
                    "lang": "en",
                    "value": "A flaw was found in postgresql. Using an UPDATE ... RETURNING command on a purpose-crafted table, an authenticated database user could read arbitrary bytes of server memory. The highest threat from this vulnerability is to data confidentiality."
                }
            ],
            "problemTypes": [
                {
                    "descriptions": [
                        {
                            "cweId": "CWE-200",
                            "description": "CWE-200",
                            "lang": "en",
                            "type": "CWE"
                        }
                    ]
                }
            ],
            "providerMetadata": {
                "dateUpdated": "2021-11-12T08:06:42",
                "orgId": "53f830b8-0a3f-465b-8143-3b8a9948e749",
                "shortName": "redhat"
            },
            "references": [
                {
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://bugzilla.redhat.com/show_bug.cgi?id=1956883"
                },
                {
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://www.postgresql.org/support/security/CVE-2021-32029/"
                },
                {
                    "tags": [
                        "x_refsource_CONFIRM"
                    ],
                    "url": "https://security.netapp.com/advisory/ntap-20211112-0003/"
                }
            ],
            "x_legacyV4Record": {
                "CVE_data_meta": {
                    "ASSIGNER": "secalert@redhat.com",
                    "ID": "CVE-2021-32029",
                    "STATE": "PUBLIC"
                },
                "affects": {
                    "vendor": {
                        "vendor_data": [
                            {
                                "product": {
                                    "product_data": [
                                        {
                                            "product_name": "postgresql",
                                            "version": {
                                                "version_data": [
                                                    {
                                                        "version_value": "postgresql 13.3, postgresql 12.7, postgresql 11.12"
                                                    }
                                                ]
                                            }
                                        }
                                    ]
                                },
                                "vendor_name": "n/a"
                            }
                        ]
                    }
                },
                "data_format": "MITRE",
                "data_type": "CVE",
                "data_version": "4.0",
                "description": {
                    "description_data": [
                        {
                            "lang": "eng",
                            "value": "A flaw was found in postgresql. Using an UPDATE ... RETURNING command on a purpose-crafted table, an authenticated database user could read arbitrary bytes of server memory. The highest threat from this vulnerability is to data confidentiality."
                        }
                    ]
                },
                "problemtype": {
                    "problemtype_data": [
                        {
                            "description": [
                                {
                                    "lang": "eng",
                                    "value": "CWE-200"
                                }
                            ]
                        }
                    ]
                },
                "references": {
                    "reference_data": [
                        {
                            "name": "https://bugzilla.redhat.com/show_bug.cgi?id=1956883",
                            "refsource": "MISC",
                            "url": "https://bugzilla.redhat.com/show_bug.cgi?id=1956883"
                        },
                        {
                            "name": "https://www.postgresql.org/support/security/CVE-2021-32029/",
                            "refsource": "MISC",
                            "url": "https://www.postgresql.org/support/security/CVE-2021-32029/"
                        },
                        {
                            "name": "https://security.netapp.com/advisory/ntap-20211112-0003/",
                            "refsource": "CONFIRM",
                            "url": "https://security.netapp.com/advisory/ntap-20211112-0003/"
                        }
                    ]
                }
            }
        }
    },
    "cveMetadata": {
        "assignerOrgId": "53f830b8-0a3f-465b-8143-3b8a9948e749",
        "assignerShortName": "redhat",
        "cveId": "CVE-2021-32029",
        "datePublished": "2021-10-08T16:44:22",
        "dateReserved": "2021-05-04T00:00:00",
        "dateUpdated": "2021-11-12T08:06:42",
        "state": "PUBLISHED"
    },
    "dataType": "CVE_RECORD",
    "dataVersion": "5.0"
}